addOns/graphql/src/main/java/org/zaproxy/addon/graphql/GraphQlGenerator.java

```diff
@@ -0,0 +1,329 @@
+            indent(query, depth);
+            query.append("}\n");
+        } else if (type instanceof GraphQLInterfaceType) {
+            List<GraphQLObjectType> objects =
+                    schema.getImplementations((GraphQLInterfaceType) type);
+            query.append("{\n");
+            for (GraphQLObjectType object : objects) {
+                indent(query, depth + 1);
+                query.append("... on ").append(object.getName()).append(' ');
+                generate(query, object, depth + 1, requestor);
+            }
+            indent(query, depth);
+            query.append("}\n");
+        } else if (type instanceof GraphQLUnionType) {
+            GraphQLUnionType union = (GraphQLUnionType) type;
+            List<GraphQLNamedOutputType> members = union.getTypes();
+            query.append("{\n");
+            for (GraphQLNamedOutputType member : members) {
+                indent(query, depth + 1);
+                query.append("... on ").append(member.getName()).append(' ');
+                generate(query, member, depth + 1, requestor);
+            }
+            indent(query, depth);
+            query.append("}\n");
+        }
+    }
+
+    private void indent(StringBuilder query, int n) {
+        for (int i = 0; i < n; ++i) {
+            query.append("    ");
+        }
+    }
+
+    private void addArguments(StringBuilder query, GraphQLFieldDefinition field) {
+        List<GraphQLArgument> args = field.getArguments();
+        if (!args.isEmpty()) {
+            query.append('(');
+            for (GraphQLArgument arg : args) {
+                query.append(arg.getName())
+                        .append(": ")
+                        .append(getDefaultValue(arg.getType()))
+                        .append(", ");
+            }
+            query.setLength(query.length() - 2);
+            query.append(") ");
+        }
+    }
+
+    private String getDefaultValue(GraphQLType type) {
+        return getDefaultValue(type, 0);
+    }
+
+    private String getDefaultValue(GraphQLType type, int depth) {
+        if (depth > MAX_DEPTH) return "null";
+        StringBuilder defaultValue = new StringBuilder();
+        if (type instanceof GraphQLNonNull) {
+            GraphQLNonNull nonNullType = (GraphQLNonNull) type;
+            type = nonNullType.getWrappedType();
+        }
+        if (type instanceof GraphQLScalarType) {
+            GraphQLScalarType scalar = (GraphQLScalarType) type;
+            switch (scalar.getName()) {
+                case "Int":
+                case "ID":
+                    defaultValue.append("1");
+                    break;
+                case "Float":
+                    defaultValue.append("3.14");
+                    break;
+                case "String":
+                    defaultValue.append("\"hello\"");
+                    break;
+                case "Boolean":
+                    defaultValue.append("true");
+                    break;
+            }
```

Comment:
Are the case strings part of a spec/standard or will we potentially run into case sensitivity problems with this?

Reply:
They are part of the spec: http://spec.graphql.org/draft/#sec-Scalars.Built-in-Scalars

Reply:
Okay great.